### ISSUE TITLE
RadioButtons control initiates unexpected calls to IValueConverter.ConvertBack

### ISSUE THREAD

**Issue by Kinnara** [microsoft/microsoft-ui-xaml#2292]
**Describe the bug**
RadioButtons control initiates unexpected additional calls to IValueConverter.ConvertBack with first param value (selected radio button index) equals '-1'.

**Steps to reproduce the bug**
1. Clone or download [the test project](https://github.com/Kinnara/RadioButtonsTest) and open MainPage.xaml.cs.
2. Set the breakpoint in ConvertBack function.
3. Debug the project and click on radio buttons. Some clicks cause ConvertBack() to be called twice. The first call is unexpected with 'value' param set to '-1'. The second call is always correct.

**Expected behavior**
No unexpected additional calls to IValueConverter.ConvertBack.

**Version Info**
NuGet package version: 
Microsoft.UI.Xaml 2.3.200213001

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  | Yes   |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Comment by deanchalk** [comment]
I had a quick look at this out of curiosity
When the initial binding is setup, we get a hit on the Convert method with the current index (0), which is expected (Radio 1 is selected by default).
When you change to the Radio 2 button, the first button Radio 1 is deselected within the C++ source code and there are no buttons selected, so a -1 value is sent to ConvertBack. Then the second Radio is selected (within the controls C++ source code) so a value  of 1 is sent to ConvertBack.
Then , if you click on Radio 1 again, we go through the same thing, -1 is sent to the ConvertBack when Radio 2 is deselected, and then 0 is sent to ConvertBack the Radio 1 is selected.

It seems like reasonable behaviour to me.

**Comment by ranjeshj** [comment]
@Kinnara can you describe what you are trying to achieve ? If you do a two way binding, I would expect convertback to be called. Is the issue that when switching from one item to the next, RadioButtons.SelectedIndex goes temporarily to -1 ?

**Comment by Kinnara** [comment]
> Is the issue that when switching from one item to the next, RadioButtons.SelectedIndex goes temporarily to -1 ?

Yes. Other selectors like ListBox or ListView (or even the original ListView-based RadioButtons, I beleive) don't have this intermediate step, and it would be reasonable for one to expect RadioButtons doesn't either. It certainly can be worked around in app code but it would be nice if RadioButtons can be changed to match other selectors in behavior. BTW the use of a converter is not really relevant and I should've simplified the test project to remove it (the issue was copied from another repo).

**Comment by ranjeshj** [comment]
Got it. We should avoid going to -1 temporarily. It is likely because we do that when an item gets unchecked [here.](https://github.com/microsoft/microsoft-ui-xaml/blob/1a39aa787427b18264b4e154cb8c7225fea6fdb3/dev/RadioButtons/RadioButtons.cpp#L420) . @StephenLPeters do you foresee any issues if we remove the intermediate -1 ? 

@Kinnara  Thanks for filing this issue. Would you like to contribute a fix ?

**Comment by Kinnara** [comment]
Sure, I'll try to find some time for it next week.

**Comment by zhuxb711** [comment]
This issue is still there in WinUI3
